### PR TITLE
GERONIMO-6844 - Migrate to Jakarta Namespace

### PR DESCRIPTION
**Pull request by rzo1**
Hi all,

don't know, what the future will be, but I did switch from Javax to Jakarta namespace (for EE10)  and updated the related spec versions. I don't know, what the plan for Geronimo provided spec jars will be, so for now, I switched to the API jars provided by the eclipse foundation.

Did switch to 4.0.0-SNAPSHOT to indicate namespace switch; would be neat to switch to Java 11 but that seems to be blocked by the Genesis parent, so likely there is the need for a Java 11 parent?

I also didn't run the sigtests / tcks yet, still fiddling around on how to set them up locally and don't know, what the general status for them looks like. 

Changes between EE9 to EE10 are (at a first glance) pretty minor for these two

- https://jakarta.ee/specifications/connectors/2.1/
- https://jakarta.ee/specifications/transactions/2.0/

But any thoughts / ideas are welcome.

### CODE REVIEW

**Comment by rzo1** [comment]
Genesis Parent: https://issues.apache.org/jira/browse/GERONIMO-6845

File: pom.xml [hunk 1]
```diff
@@ -62,45 +62,45 @@
...
             </dependency>
             <dependency>
                 <groupId>org.apache.geronimo.components</groupId>
                 <artifactId>geronimo-connector</artifactId>
-                <version>3.1.6-SNAPSHOT</version>
+                <version>4.0.0-SNAPSHOT</version>
             </dependency>
 
             <!-- external -->
             <dependency>
```
Comment (rmannibucau) [inline pom.xml]: think the new major is the opportunity to ditch this required dep and migrate to jul for consistency with the other stacks and new API.

**Comment by rmannibucau** [comment]
As mentionned in the ticket think we don't need GERONIMO-6845, however we need to validate we use jakarta artifact and complete the migration of spec jars before being able to merge the change.

Except the slf4j and this "move task" +1 from me once they are both done.

File: pom.xml (hunk 1 above)
Comment (rzo1) [reply pom.xml]: I am fine to migrate slf4j to jul.

**Comment by rzo1** [comment]
> As mentionned in the ticket think we don't need GERONIMO-6845, however we need to validate we use jakarta artifact and complete the migration of spec jars before being able to merge the change.

I am fine with either using the java11 flava thing or apache-parent directly. Whatever the project wants to use.

Yup - there is an ongoing discussion on the list regarding the spec jars, if I remember correctly.

File: pom.xml (hunk 1 above)
Comment (rzo1) [reply pom.xml]: done

**Comment by rzo1** [comment]
Ok did the switch to Apache Parent v29

**Comment by jeanouii** [comment]
Looks good to me. I'll merge it for you

File: pom.xml [hunk 2]
```diff
@@ -113,102 +95,70 @@
...
-        </profile>
-    </profiles>
     <build>
         <pluginManagement>
             <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <configuration>
+                        <source>11</source>
```
Comment (jeanouii) [inline pom.xml]: I'd use the new release attribute

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        java: [ 11 ]
```
Comment (jeanouii) [inline .github/workflows/maven.yml:27]: I'd create a ticket for this component to mention the Java 11 version as being the minimum required version from now on so it appears in the release notes

File: geronimo-connector/src/main/java/org/apache/geronimo/connector/outbound/AbstractSinglePoolConnectionInterceptor.java [hunk 4]
```diff
@@ -24,21 +24,20 @@
...
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 /**
  * @version $Rev$ $Date$
  */
 public abstract class AbstractSinglePoolConnectionInterceptor implements ConnectionInterceptor, PoolingAttributes {
-    protected static Logger log = LoggerFactory.getLogger(AbstractSinglePoolConnectionInterceptor.class);
+    protected static Logger log = Logger.getLogger(AbstractSinglePoolConnectionInterceptor.class.getName());
```
Comment (jeanouii) [inline geronimo-connector/src/main/java/org/apache/geronimo/connector/outbound/AbstractSinglePoolConnectionInterceptor.java:40]: I'd also make a small ticket for this component to have the logger switch to JUL

File: pom.xml [hunk 5]
```diff
@@ -113,102 +95,70 @@
...
                     <executions>
                         <execution>
                             <id>attach-javadocs</id>
                             <goals>
                                 <goal>jar</goal>
                             </goals>
                             <configuration>
-                                <additionalparam>${javadoc.opts}</additionalparam>
+                                <quiet>true</quiet>
+                                <source>11</source>
```
Comment (jeanouii) [inline pom.xml]: I think there is a release attribute as well

File: geronimo-connector/src/main/java/org/apache/geronimo/connector/outbound/AbstractSinglePoolConnectionInterceptor.java (hunk 4 above)
Comment (jeanouii) [reply geronimo-connector/src/main/java/org/apache/geronimo/connector/outbound/AbstractSinglePoolConnectionInterceptor.java:40]: Haven't seen SLF4J dependency removed. Maybe we were using a transitive dependency as compile

**Review by jeanouii — CHANGES_REQUESTED** [review]
Do you mind doing the small tweaks or I can merge and do it myself afterwards

File: .github/workflows/maven.yml (hunk 3 above)
Comment (rzo1) [reply .github/workflows/maven.yml:27]: https://issues.apache.org/jira/browse/GERONIMO-6847

**Pull request merged**